packages/application-config/src/validations.ts

```diff
@@ -45,4 +47,16 @@ const validateConfig = (
   }
 };
 
-export default validateConfig;
+export const validateSubmenuLinks = (
+  submenuLinks: CustomApplicationData['submenuLinks']
```

Comment:
Nit: maybe you can apply the changes here as well, to pass the Custom App object instead of the submenu links (same as in the services)

Reply:
updated